### PR TITLE
Make top-level XML parsing more robust

### PR DESCRIPTION
**Pull request by filmackay**
This PR introduces proper XML (rather than string) parsing, so any compliant `.dc` file will work - not just ones that happen to have whitespace in the right places. XML queries have also been made more specific, as existing `.//Objects/*` queries were too global and TPT screens would incorrectly match on them.

### CODE REVIEW

File: pyvantage/__init__.py [hunk 1]
```diff
@@ -1369,12 +1365,14 @@ def load_xml_db(self, disable_cache=False, config_dir="./"):
...
             response = response.decode('ascii')
-            response = response[response.find("</Result>\n")+10:]
-            response = response.replace('<?File Encode="Base64" /', '')
-            response = response.replace('?>', '')
-            response = response[:response.find('</return>')]
-            dbytes = base64.b64decode(response)
-            xml_db = dbytes.decode('utf-8')
+
+            # read XML preserving processing instructions
+            from xml.etree import ElementTree as ET
```
Comment (loopj) [inline pyvantage/__init__.py]: Maybe we should move this import (and the one from `parse`) to the top now?

**Comment by loopj** [comment]
I think this PR is solid. Tested on my system at home.

**Comment by filmackay** [comment]
@loopj this is ready to merge if you are happy with it.

**Comment by gjbadros** [comment]
Thanks @filmackay and @loopj  -- I'm just doing some diligence on the change of XPath expressions.

**Review by gjbadros — APPROVED** [review]
Ok, this looks good... I've not tested on my system yet, but will after I merge this (and just merged a bunch of other PRs, too, so I can just test once-ish).

**Pull request merged**